src/main/java/io/oneko/kubernetes/impl/DeploymentStatusWatcher.java

```diff
@@ -74,20 +70,22 @@ private boolean shouldScanDeployable(ProjectVersion<?, ?> projectVersion) {
 	}
 
 	private void scanResourcesForDeployable(ProjectVersion<?, ?> projectVersion) {
-		try {
-			final List<Status> statuses = HelmCommandUtils.status(projectVersion);
+		projectVersionLock.doWithProjectVersionLock(projectVersion, () -> {
```

Comment:
Will this work well without a lot of blocking time? The scheduled task is invoked every 5 seconds.

Reply:
Yes, this will cause some blocking time. But the `scanResourcesForDeployable` Method is called multiple times per update and therefore releases the lock quite often.

Reply:
I'll increase the interval time to 10s.